src/common.hh

```diff
@@ -21,7 +43,38 @@ struct ImageDims
     }
 };
 
+struct Zarr;
+
 namespace common {
+struct ThreadPool final
+{
+  public:
+    using JobT = std::function<bool(std::string&)>;
+    struct ThreadContext
+    {
+        std::thread thread;
+        std::mutex mutex;
+        std::condition_variable cv;
+        bool should_stop;
+        bool ready;
+    };
+
+    ThreadPool(size_t n_threads, std::function<void(const std::string&)> err);
```

Comment:
Need to document the error handler by adding a comment here. When's it called and what is the recommended behavior? Any constraints?